src/oracle_implementations/discount_rate/NotionalFinance/NotionalFinanceValueProvider.sol

```diff
@@ -112,4 +105,31 @@ contract NotionalFinanceValueProvider is Oracle, Convert {
         // The result is a 59.18 fixed-point number.
         return discreteRateD59x18;
     }
+
+    /// @notice Retrieve the oracle rate from the NotionalFinance Market
+    function getOracleRate() internal view returns (uint256) {
+        uint256 settlementDate = getSettlementDate();
+        // Attempt to retrieve the oracle rate directly by using the maturity and settlement date
+        MarketParameters memory marketParams = INotionalView(notionalView)
+            .getMarket(uint16(currencyId), maturityDate, settlementDate);
+
+        // If we find an active market we can return the oracle rate otherwise we revert
+        if (marketParams.oracleRate <= 0) {
+            revert NotionalFinanceValueProvider__getValue_invalidMarketParameters(
+                currencyId,
+                maturityDate,
+                settlementDate
+            );
+        }
+
+        return marketParams.oracleRate;
+    }
+
+    /// @notice Computes the settlement date as is done in the NotionalFinance contracts
+    /// Reference 1: https://github.com/notional-finance/contracts-v2/blob/master/contracts/internal/markets/DateTime.sol#L14
```

Comment:
Better to use absolute links (include commit hash instead of branch) since that will change in the future and it will point to a different line